src/webgpu/api/validation/createBindGroupLayout.spec.ts

```diff
@@ -88,6 +94,96 @@ g.test('mininum_buffer_binding_size')
     }, !success);
   });
 
+g.test('view_dimension')
+  .params(
+    params()
+      .combine(poptions('type', kBindingTypes))
+      .combine(poptions('viewDimension', [undefined, '2d', 'cube'] as const))
+  )
+  .fn(async t => {
+    const { type, viewDimension } = t.params;
+
+    let success = false;
+    if (
+      viewDimension === undefined ||
+      (viewDimension in kTextureViewDimensionInfo && type in kTextureBindingTypeInfo)
+    ) {
+      success = true;
+    }
+
+    t.expectValidationError(() => {
+      t.device.createBindGroupLayout({
+        entries: [{ binding: 0, visibility: GPUShaderStage.COMPUTE, type, viewDimension }],
+      });
+    }, !success);
+  });
+
+g.test('texture_component_type')
+  .params(
+    params()
+      .combine(poptions('type', kBindingTypes))
+      .combine(poptions('textureComponentType', [undefined, 'float', 'uint'] as const))
+  )
+  .fn(async t => {
+    const { type, textureComponentType } = t.params;
+
+    let success = false;
+    if (
+      textureComponentType === undefined ||
+      (textureComponentType in kTextureComponentTypeInfo && type === 'sampled-texture')
```

Comment:
same, I think `textureComponentType in kTextureComponentTypeInfo` is always true

Reply:
OK. removed this